### PR TITLE
Fix the root motion extraction rotation problem

### PR DESCRIPTION
**Pull request by amzn-rhhong**
Fixed the problem that when rotation is enabled the character feet sink under the ground.
Before:
image
After:
image
https://user-images.githubusercontent.com/69218254/181704634-c0de6960-8b5d-4277-af24-827b01350f36.mp4

Also added some debug rendering for root motion

### CODE REVIEW

File: Gems/EMotionFX/Code/EMotionFX/Source/MotionData/NonUniformMotionData.cpp [hunk 1]
```diff
@@ -1238,12 +1238,21 @@ namespace EMotionFX
...
+
+                    // Final root rotation only keeps the rotation around Z-axis
+                    const AZ::Vector3 rootRotEular = AZ::ConvertQuaternionToEulerRadians(sampleJointRot);
+                    const AZ::Quaternion finalRootRot = AZ::ConvertEulerRadiansToQuaternion(AZ::Vector3(0, 0, rootRotEular.GetZ()));
+
+                    // Calculate the final sample rotation
+                    AZ::Quaternion finalSampleRot = finalRootRot.GetInverseFull() * sampleJointRot;
+
+                    m_jointData[rootJointDataIndex].m_rotationTrack.m_values[i].FromQuaternion(finalRootRot);
+                    m_jointData[sampleJointDataIndex].m_rotationTrack.m_values[i].FromQuaternion(finalSampleRot);
```
Comment (amzn-jillich) [inline Gems/EMotionFX/Code/EMotionFX/Source/MotionData/NonUniformMotionData.cpp:1255]: Is this the code that is being called at asset processor time?

Comment (amzn-jillich) [inline Gems/EMotionFX/Code/EMotionFX/Source/MotionData/NonUniformMotionData.cpp:1255]: If you want to remove the X and Y rotation from a quaternion you can also do this:

// Extract only Z rotation by zeroing out X and Y and then renormalizing the quaternion.
```
rot.x = 0.0f;
rot.y = 0.0f;
rot.Normalize();
```

**Review by amzn-jillich — APPROVED** [review]

File: Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp [hunk 2]
```diff
@@ -1209,4 +1214,20 @@ namespace AZ::Render
             oldRight = vertices[3];
         }
     }
+
+    void AtomActorDebugDraw::RenderRootMotion(AzFramework::DebugDisplayRequests* debugDisplay,
```
Comment (hultonha) [inline Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp:1218]: Always love debug rendering! 😄

File: Gems/EMotionFX/Code/EMotionFX/Source/MotionData/NonUniformMotionData.cpp [hunk 3]
```diff
@@ -1238,12 +1238,21 @@ namespace EMotionFX
...
                 m_jointData[sampleJointDataIndex].m_positionTrack.m_values[i].Set(x, y, z);
 
                 if (extractRotation)
                 {
-                    m_jointData[sampleJointDataIndex].m_rotationTrack.m_values[i].FromQuaternion(AZ::Quaternion::CreateIdentity());
+                    const AZ::Quaternion sampleJointRot = m_jointData[sampleJointDataIndex].m_rotationTrack.m_values[i].ToQuaternion();
+
+                    // Final root rotation only keeps the rotation around Z-axis
+                    const AZ::Vector3 rootRotEular = AZ::ConvertQuaternionToEulerRadians(sampleJointRot);
+                    const AZ::Quaternion finalRootRot = AZ::ConvertEulerRadiansToQuaternion(AZ::Vector3(0, 0, rootRotEular.GetZ()));
```
Comment (hultonha) [inline Gems/EMotionFX/Code/EMotionFX/Source/MotionData/NonUniformMotionData.cpp:1249]: nit/suggestion: Might be worth updating `Rot` to `Rotation` just to make the variable names a little clearer

File: Gems/EMotionFX/Code/EMotionFX/Source/MotionData/NonUniformMotionData.cpp [hunk 4]
```diff
@@ -1238,12 +1238,21 @@ namespace EMotionFX
...
-
                 m_jointData[sampleJointDataIndex].m_positionTrack.m_values[i].Set(x, y, z);
 
                 if (extractRotation)
                 {
-                    m_jointData[sampleJointDataIndex].m_rotationTrack.m_values[i].FromQuaternion(AZ::Quaternion::CreateIdentity());
+                    const AZ::Quaternion sampleJointRot = m_jointData[sampleJointDataIndex].m_rotationTrack.m_values[i].ToQuaternion();
+
+                    // Final root rotation only keeps the rotation around Z-axis
+                    const AZ::Vector3 rootRotEular = AZ::ConvertQuaternionToEulerRadians(sampleJointRot);
```
Comment (hultonha) [inline Gems/EMotionFX/Code/EMotionFX/Source/MotionData/NonUniformMotionData.cpp:1248]: nit: Eular -> Euler

File: Gems/EMotionFX/Code/EMotionFX/Source/MotionData/NonUniformMotionData.cpp [hunk 5]
```diff
@@ -1238,12 +1238,21 @@ namespace EMotionFX
...
+                    const AZ::Quaternion finalRootRot = AZ::ConvertEulerRadiansToQuaternion(AZ::Vector3(0, 0, rootRotEular.GetZ()));
+
+                    // Calculate the final sample rotation
+                    AZ::Quaternion finalSampleRot = finalRootRot.GetInverseFull() * sampleJointRot;
```
Comment (hultonha) [inline Gems/EMotionFX/Code/EMotionFX/Source/MotionData/NonUniformMotionData.cpp:1252]: suggestion: This could also be `const` 🙂

**Review by hultonha — APPROVED** [review]
Updates look good to me, only a couple of small comments. One other question is do we think there's a way we could test this to validate the before/after case? Would definitely be nice to capture this in a test and help prevent any regressions

File: Gems/EMotionFX/Code/EMotionFX/Source/MotionData/NonUniformMotionData.cpp (hunk 1 above)
Comment (amzn-rhhong) [reply Gems/EMotionFX/Code/EMotionFX/Source/MotionData/NonUniformMotionData.cpp:1255]: Yes, the root motion extraction process happens at AP time.

**Pull request merged**